src/jLouvain.js

```diff
@@ -83,11 +83,12 @@ function get_graph_size(graph) {
 function add_edge_to_graph(graph, edge) {
 	update_assoc_mat(graph, edge);
 
-	if (edge_index[edge.source + '_' + edge.target]) {
-		graph.edges[edge_index[edge.source + '_' + edge.target]].weight = edge.weight;
+	let key = edge.source + '_' + edge.target;
+	if (edge_index.hasOwnProperty(key)) {
+		graph.edges[edge_index[key]].weight = edge.weight;
 	} else {
+		edge_index[key] = graph.edges.length;
 		graph.edges.push(edge);
-		edge_index[edge.source + '_' + edge.target] = graph.edges.length - 1;
```

Comment:
I don't quite understand why you think this is wrong? Indexes shouldn't go up to length; otherwise the other changes look fine :)

Reply:
Oh, this logic is not wrong at all. It's just that subtracting 1 seemed a bit redundant since we can get the same value from `graph.edges.length` prior to adding the edge to `graph.edges`.